_msprimemodule.c

```diff
@@ -122,9 +122,9 @@ typedef struct {
 typedef struct {
     PyObject_HEAD
     SparseTree *sparse_tree;
-    leaf_list_node_t *head;
-    leaf_list_node_t *tail;
-    leaf_list_node_t *next;
+    node_list_t *head;
+    node_list_t *tail;
+    node_list_t *next;
 } LeafListIterator;
```

Comment:
By my reading this should be `SampleListIterator`, but this is never externally visible, so not a big deal.